llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp

```diff
@@ -449,7 +449,8 @@ struct AAAMDAttributesFunction : public AAAMDAttributes {
     // Check for Intrinsics and propagate attributes.
     const AACallEdges *AAEdges = A.getAAFor<AACallEdges>(
         *this, this->getIRPosition(), DepClassTy::REQUIRED);
-    if (!AAEdges || AAEdges->hasNonAsmUnknownCallee())
+    if (!AAEdges || !AAEdges->isValidState() ||
```

Comment:
I'm still not sure when these getAAFors can actually fail

Reply:
If the IR position is not valid for the queried AA, it can fail.

Reply:
But for all of these function only attributes, how could that happen?

Reply:
For those all the AAs here (in this file), it might not fail. There are generally three cases: 1) invalid IR position 2) filtered out AA (not in allow list) 3) function AA with `optnone`.